### ISSUE TITLE
[Question] Which learning rate for Training?

### ISSUE THREAD

**Issue by CaptainDario** [sebastian-sz/efficientnet-lite-keras#31]
I Am trying to train EfficientNet B0 on my dataset of ~10.000.000 images of shape (128, 128, 1) with 6500 classes.
However I get a very weird fluctuating validation loss / accuracy. I am not sure what the problem is but guessing it is related to a wrong learning rate. So I am wondering what learning rate to choose.

**Comment by sebastian-sz** [comment]
Hello, it is hard to say why exactly is the loss behaving like that. I have few questions:
1. This dataset seems very big. Are you training from scratch or using pretrained weights (transfer learning) ?
2. If you train only on 1 single image for ~100 epochs does your loss also fluctuate like that? (you should easily get 100% accuracy on a single image).
3. What is your starting learning rate?

**Comment by CaptainDario** [comment]
Thank you for your quick reply.

1. I am training from scratch, mainly because I am using grayscale images.
2. I can try this today
3. I didn't specify a learning rate and I am using the default for adam

**Comment by CaptainDario** [comment]
I tried training it on only 50 of the classes. The history looks like this:

![grafik](https://user-images.githubusercontent.com/51273483/158017656-2851291b-eeed-43d7-bb05-dfe568950af0.png)

Is this related to the learning rate?

**Comment by sebastian-sz** [comment]
@CaptainDario Sorry I am not sure. I don't have much experience in training models from scratch on such large datasets.

**Comment by CaptainDario** [comment]
@sebastian-sz This took a long time but I figured it out. After reading the EfficientNet paper I knew which hyperparameters, especially the learning rate to use.
So the key to solving this issue was to decay the learning rate by 0.97 every 2.4 epochs.

**Comment by sebastian-sz** [comment]
@CaptainDario Glad you found it!
